libcloud/dns/drivers/godaddy.py

```diff
@@ -0,0 +1,521 @@
+    connectionCls = GoDaddyDNSConnection
+
+    RECORD_TYPE_MAP = {
+        RecordType.A: 'A',
+        RecordType.AAAA: 'AAAA',
+        RecordType.CNAME: 'CNAME',
+        RecordType.MX: 'MX',
+        RecordType.NS: 'SPF',
+        RecordType.SRV: 'SRV',
+        RecordType.TXT: 'TXT',
+    }
+
+    def __init__(self, shopper_id, key, secret,
+                 secure=True, host=None, port=None):
+        """
+        Instantiate a new `GoDaddyDNSDriver`
+
+        :param  shopper_id: Your customer ID or shopped ID with GoDaddy
+        :type   shopper_id: `str`
+
+        :param  key: Your access key from developer.godaddy.com
+        :type   key: `str`
+
+        :param  secret: Your access key secret
+        :type   secret: `str`
+        """
+        super(GoDaddyDNSDriver, self).__init__(key=key, secret=secret,
+                                               secure=secure,
+                                               host=host, port=port,
+                                               shopper_id=str(shopper_id))
+
+    def _to_zones(self, items):
+        zones = []
+        for item in items:
+            zones.append(self._to_zone(item))
+        return zones
+
+    def _to_zone(self, item):
+        extra = {"expires": item['expires']}
+        zone = Zone(id=item['domainId'], domain=item['domain'],
+                    type='master', ttl=None,
+                    driver=self, extra=extra)
+        return zone
+
+    def _to_records(self, items, zone=None):
+        records = []
+
+        for item in items:
+            records.append(self._to_record(item=item, zone=zone))
+        return records
+
+    def _to_record(self, item, zone=None):
+        ttl = item['ttl']
+        type = self._string_to_record_type(item['type'])
+        name = item['name']
+        id = '{0}:{1}'.format(name, type)
+        record = Record(id=id, name=name,
+                        type=type, data=item['data'],
+                        zone=zone, driver=self,
+                        extra={'ttl': ttl})
+        return record
+
+    def _to_tlds(self, items):
+        tlds = []
+        for item in items:
+            tlds.append(self._to_tld(item))
+        return tlds
+
+    def _to_tld(self, item):
+        return GoDaddyTLD(
+            name=item['name'],
+            tld_type=item['type']
+        )
+
+    def list_zones(self):
+        """
+        Return a list of zones (purchased domains)
+
+        :return: ``list`` of :class:`Zone`
+        """
+        result = self.connection.request(
+            '/v1/domains/').object
+        zones = self._to_zones(result)
+        return zones
+
+    def list_records(self, zone):
+        """
+        Return a list of records for the provided zone.
+
+        :param zone: Zone to list records for.
+        :type zone: :class:`Zone`
+
+        :return: ``list`` of :class:`Record`
+        """
+        result = self.connection.request(
+            '/v1/domains/%s/records' % zone.domain).object
```

Comment:
Minor code style / safety thing - always when using string interpolation wrap arguments in parenthesis - it prevents accidental errors which could arise if user adds another argument, but forgets to add parenthesis (we had those issues in production before, ideally tests should discover that, but still).